xknx/io/knxip_interface.py

```diff
@@ -162,3 +166,33 @@ def telegram_received(self, telegram):
     async def send_telegram(self, telegram):
         """Send telegram to connected device (either Tunneling or Routing)."""
         await self.interface.send_telegram(telegram)
+
+    def find_local_ip(self, gateway_ip):
+        """Find local IP address on same subnet as gateway."""
+        def _scan_interfaces(gateway: ipaddress.IPv4Address):
+            for interface in netifaces.interfaces():
+                try:
+                    af_inet = netifaces.ifaddresses(interface)[netifaces.AF_INET]
+                    for link in af_inet:
+                        network = ipaddress.IPv4Network((link["addr"],
+                                                         link["netmask"]),
+                                                        strict=False)
+                        if gateway in network:
+                            self.xknx.logger.debug("Using interface: %s", interface)
+                            return link["addr"]
+                except KeyError:
+                    self.xknx.logger.debug("Could not find IPv4 address on interface %s", interface)
+                    continue
+            raise XKNXException("No interface on same subnet as gateway found.")
```

Comment:
I think we should not raise an exception here - and work with `None`.

Reply:
The intention was to raise once -> try default gateway. Raise twice (no  default gateway found) -> exit since a bind(None) will for sure fail, but without a good error message.

Reply:
:)